### DISCUSSION TITLE
Add class to checkbox when using TaskListExtension & DefaultAttributesExtension?

### DISCUSSION THREAD

**Discussion by TaromaruYuki** [thephpleague/commonmark#947] [Q&A]
Hello! I have been following along with #605, specifically [this reply](https://github.com/thephpleague/commonmark/issues/605#issuecomment-1011531334). I can get everything but the input checkbox itself.

```php
ListBlock::class => [
    'class' => static function (ListBlock $listblock) {
        if ($listblock->firstChild()->firstChild()->firstChild() instanceof TaskListItemMarker) {
            return 'bg-primary';
        }

        return null;
    },
],
ListItem::class => [
    'class' => static function (ListItem $listitem) {
        if ($listitem->firstChild()->firstChild() instanceof TaskListItemMarker) {
            return 'bg-danger';
        }

        return null;
    },
]
```

Output:
![image](https://user-images.githubusercontent.com/48397556/199380837-fcee812d-4bfc-4b10-b5fb-2cdc260be1bd.png)

![image](https://user-images.githubusercontent.com/48397556/199380894-d2495a40-7e2c-4cb7-bfc2-1ff79f6dd4ce.png)

Is it possible to add a class to the input element itself so I can add Bootstrap styling to it?

**Comment by colinodell** [comment]
That checkbox comes from this node type: https://github.com/thephpleague/commonmark/blob/2.3/src/Extension/TaskList/TaskListItemMarker.php Try adding a third entry to your config for `TaskListItemMarker` and see if that works.

**Reply by TaromaruYuki** [reply]
Does not work 🤔 

I am using symfony's dd function and I do get a dump of the task list. So, it is being found, but applying classes to it does not do anything at all.

![image](https://user-images.githubusercontent.com/48397556/199527652-cfcd30ae-f931-4297-9b5a-43ae7a8cd0dd.png)

![image](https://user-images.githubusercontent.com/48397556/199527887-6a787a63-548d-4f9e-9d76-e8f57f661a8b.png)

Doing some more testing, applying anything at all does not work. For example, giving it an id and searching for the id via JavaScript returns null. So, is TaskListItemMarker ignoring the config?

![image](https://user-images.githubusercontent.com/48397556/199527319-15da0703-27f0-48f3-b2d9-89b111cf624c.png)

**Reply by colinodell** [reply]
It looks like there's a bug in the class that renders the checkboxes - it's not including any of the attributes set by `DefaultAttributesExtension`.  I'll prepare a fix for this now!

**Reply by TaromaruYuki** [reply]
Thank you! I swear I thought I was doing something wrong 😄

**Reply by TaromaruYuki** [reply]
Trying to target `ListItem` to add a class to the li's of the table of contents also does not work. `TableOfContents` also does not work. 🤔
